File: 02-week-2--recursion/03-day-3--recursive-fibonacci-series/javascript/fibonacci_recursive.js
function fibonacci(n) {
  if ( n < 2 ) {
    return n;
  } else {
    
  }
  
  const array = [0, 1]
  let sum;  
  sum = array[0] + array[1];
  array[0] = array[1];
  array[1] = sum;

  if (n === 0) {
    return array[0]
  }

  fibonacci(n-1);
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));

  console.log("Expecting: 1");
  console.log("=>", fibonacci(3));

  console.log("Expecting: 8");
  console.log("=>", fibonacci(7));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
  //check if argument is 0, 1, or 2, those will always be 0, 1,1 
  //make array of 2
  //sum the values of the array
  //set array[0] === to array[1] and array[1] === sum
  //loop through number of times equal to argument.
  //return sum


// And a written explanation of your solution

File: 02-week-2--recursion/03-day-3--recursive-fibonacci-series/ruby/fibonacci_recursive.rb
def fibonacci(n)
  if n < 2
    return n
  end

  values = [0,1]

  (n-1).times do 
    values << values[-1] + values[-2]
  end

  values.last
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  puts "=>", fibonacci(100)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution
